webapp/security/views.py

```diff
@@ -333,5 +342,208 @@ def cve(cve_id):
 # ===
 
 
-def api_create_cve():
-    pass
+def create_cve():
+    """
+    Receives a POST request from load_cve.py
+    Parses the object and bulk inserts with add_all()
+    @params json: the body of the request
+    """
+
+    data = flask.request.get_json()
+    response = flask.jsonify({"message": "Unable to get body"}), 400
+    packages = []
+    references = []
+    bugs = []
+
+    # Check if CVE exists by candidate
+    if db_session.query(CVE).filter(CVE.id == data["id"]).count() > 0:
+        response = flask.jsonify({"message": "CVE already exists"}), 400
+        return response
+
+    # Packages
+    # Check if there are packages before mapping
+    if "packages" in data and len(data["packages"]) > 0:
+        for pkg in data["packages"]:
+            releases = []
+            for rel in pkg["releases"]:
+                release = PackageReleaseStatus(
+                    name=rel["name"],
+                    status=rel["status"],
+                    status_description=rel["status_description"],
+                )
+            releases.append(release)
+            package = Package(
+                name=pkg["name"],
+                source=pkg["source"],
+                ubuntu=pkg["ubuntu"],
+                debian=pkg["debian"],
+                releases=releases,
+            )
+            packages.append(package)
+
+    if "references" in data and len(data["references"]) > 0:
+        for ref in data["references"]:
+            reference = CVEReference(uri=ref)
+            references.append(reference)
+
+    if "bugs" in data and len(data["bugs"]) > 0:
+        for b in data["bugs"]:
+            bug = Bug(uri=b)
+            bugs.append(bug)
+
+    cves = [
+        CVE(
```

Comment:
You're only adding one CVE? In which case, why make it a list?

Reply:
Because the SQLAlchemy method `add_all` at the end only accepts a list

Reply:
Umm... You use `add_all` if you have a list, you use [`add`](https://docs.sqlalchemy.org/en/13/orm/session_api.html#sqlalchemy.orm.session.Session.add) if you have a single item. Right?